lib/changelog_source.rb

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+class ChangelogSource
+
+  attr_reader :elements
+
+  def initialize(file)
+    @file = file
+    @elements = {}
+  end
+
+  def parse
+    dirty_elements = File.read(@file).split('##')
+    dirty_elements.each do |dirty_element|
+      element = parse_element(dirty_element)
+      unless element.nil?
+        version, *dirty_items = *dirty_element.split('*')
```

Comment:
Useless assignment to variable - `version`. Use `_` or `_version` as a variable name to indicate that it won't be used.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.